core/errors/errors.go

```diff
@@ -46,34 +68,53 @@ func (e Error) Error() string {
 // Format returns a formatted new error based on the arguments
 // it does NOT change the original error's message
 func (e Error) Format(a ...interface{}) Error {
-	e.message = fmt.Sprintf(e.message, a...)
+	e.Message = fmt.Sprintf(e.Message, a...)
+	return e
+}
+
+func omitNewLine(message string) string {
+	if strings.HasSuffix(message, "\n") {
```

Comment:
Have now changed it. Hope I'm right with this integration like you mean it should be done.